src/sympc/protocol/falcon/falcon.py

```diff
@@ -63,3 +69,103 @@ def __eq__(self, other: Any) -> bool:
             return False
 
         return True
+
+    @staticmethod
+    def mul_master(x: MPCTensor, y: MPCTensor, session: Session) -> MPCTensor:
+        """Master method for multiplication.
+
+        Performs Falcon's mul implementation, gets and reshares mul results and distributes shares.
+        This needs to be improved in future, it relies on orchestrator being a trusted third party.
+        Falcon, requires parties to be able to communication between each other.
+
+        Args:
+            x (MPCTensor): Secret
+            y (MPCTensor): Another secret
+            session (Session): Session the tensors belong to
+
+        Returns:
+            shares (ReplicatedSharedTensor): Shares in terms of ReplicatedSharedTensor.
+
+        Raises:
+            ValueError: Raised when number of parties are not three.
+            NotImplementedError: Raised when implementation not present
+
+        """
+        if len(session.parties) != 3:
+            raise ValueError("Falcon requires 3 parties")
+
+        result = None
+
+        if session.protocol.security_type == "semi-honest":
+            args = []
+            for index in range(0, 3):
+                args.append(
+                    [
+                        x.share_ptrs[index],
+                        y.share_ptrs[index],
+                        session.session_ptrs[index],
```

Comment:
MAYBE: You don't need to send the session - you can send only ```x and y``` (as you already do) - and in  ```Falcon.compute_zvalue_and_add_mask``` do:
```
session_uuid = x.session_uuid
session = get_session(str(session_uuid)) # get_session is in session_utils.py
```